server/memo.go

```diff
@@ -30,6 +30,19 @@ func (s *Server) registerMemoRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusBadRequest, "Malformatted post memo request").SetInternal(err)
 		}
 
+		// Find system settings
+		systemSettingKey := api.SystemSettingDisablePublicMemosName
+		systemSetting, err := s.Store.FindSystemSetting(ctx, &api.SystemSettingFind{
+			Name: &systemSettingKey,
+		})
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to find system setting").SetInternal(err)
+		}
+
+		if systemSetting.Value == "true" {
```

Comment:
Unmarshal it before using it.

e.g. https://github.com/usememos/memos/blob/main/server/memo.go#L45